src/types/actions.js

```diff
@@ -361,6 +361,7 @@ type UrlStateAction =
   | {|
       +type: 'CHANGE_SHOW_TAB_ONLY',
       +showTabOnly: BrowsingContextID | null,
+      +selectedTab: TabSlug,
```

Comment:
Not for this patch: I think I did similar approaches elsewhere when tabs disappear, but I think this logic for switching to a supported tab could be placed in a selector.